azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -933,6 +958,23 @@ func expandKubernetesClusterAgentPoolProfiles(d *schema.ResourceData) []containe
 		if vnetSubnetID != "" {
 			profile.VnetSubnetID = utils.String(vnetSubnetID)
 		}
+
+		if maxCount := int32(config["max_count"].(int)); maxCount > 0 {
+			profile.MaxCount = utils.Int32(maxCount)
+		}
+
+		if minCount := int32(config["min_count"].(int)); minCount > 0 {
+			profile.MinCount = utils.Int32(minCount)
+		}
+
+		if enableAutoScalingItf := config["enable_auto_scaling"]; enableAutoScalingItf != nil {
+			profile.EnableAutoScaling = utils.Bool(enableAutoScalingItf.(bool))
+		}
```

Comment:
there's a couple of things here:

1. if auto-scaling's enabled and we don't have a min/max set - should we raise an error here? This'd require updating the function to also support returning/raising errors:

```
func expandKubernetesClusterAgentPoolProfiles(d *schema.ResourceData) ([]containerservice.ManagedClusterAgentPoolProfile, error) {
```

(for which we can return `nil` by default for the error)

2. if auto-scaling is enabled; should we conditionally not be setting the 'count' field? I'm wondering if that will force a node-pool resizing during an update?

Reply:
Hi @tombuildsstuff 
Fully agree with 1) I'll add it to the patch.
about the 2) is more complicated: if you say { min:1, max:3, count:2 }, The cluster is created with 2 nodes. If you don't send the count you get:
```
{
    "code": "InvalidParameter",
    "message": "The value of parameter agentPoolProfile.count is invalid. Please see https://aka.ms/aks-naming-rules for more details.",
    "target": "agentPoolProfile.count"
}
```

The issue comes with the updates, despite that is the same HTTP methos and URL the api has some  really different behaviour depending the type of agent pool (not related to autoscaling).

Any update to the AKS  cluster if the node pool is  `VirtualMachineScaleSets` will throw:
```
{
    "code": "OperationNotAllowed",
    "message": "Node pool update via managed cluster not allowed. Use per nodepool operations."
}
```

As soon as that is clarified with the AKS team, then we can go left and handle the exception in terraform, or they'll make the API to behave the same regardless the type of agentpool.